### ISSUE TITLE
Wavedrom encodings are inconsistently formatted

### ISSUE THREAD

**Issue by christian-herber-nxp** [riscv/riscv-isa-manual#1430]
I am failing to understand the logic behind the formatting of instruction encodings in wavedrom.
For one, it is quite inconsistent between extensions (e.g. compare C to Zc).
But even within extensions, or even a single instruction, I do not understand how the type is assigned (that translates to the background color). See e.g.:
![image](https://github.com/riscv/riscv-isa-manual/assets/101172126/5492c960-8e8e-4517-80e2-b96282b44e11)

Two parts of one immediate have different colors. One operand red, one operand green. 

Having clarity on how to format the instructions will greatly help authors of extensions to do it right from the start.

**Comment by christian-herber-nxp** [comment]
@wmat : thanks, i updated the colors in https://github.com/riscv/riscv-zilsd/pull/18. 
From that, I can for sure say that the types in C need to be revisited.

**Comment by gfavor** [comment]
Given that the "old" ISA manuals did not use color coding, should we drop all color coding in the new manuals?  Or should we have Bill identify one standard color coding that we will gradually convert to?

**Comment by ved-rivos** [comment]
IMO: The colors are a bit distracting and don't add much. The look and feel of the "old" manual was better.

**Comment by wmat** [comment]
The addition of colored backgrounds to the wavedrom diagrams predated my
joining the project.  I always assumed it was done for a reason but perhaps
not.  Removing the colorization (type attributes) from the wavedrom would
certainly make writing wavedrom that much easier.

**Comment by gfavor** [comment]
Let's see if @aswaterman concurs with dropping all color coding.

**Comment by christian-herber-nxp** [comment]
personally, I like the colors, if used meaningfully and consistently.
Especially in less regular encodings like C/Zc, helps to get an overview.

**Comment by aswaterman** [comment]
I think the two valid resolutions are to drop the colors or to use the colors consistently.  Given the latter option is much more work, and isn't exactly of high priority, I'd be fine with dropping them.  We can always reverse course in the future if we have the bandwidth.

**Comment by wmat** [comment]
I assume the colors are simply decoration, correct?

Bill Traynor
Documentation Build and Release Engineer
RISC-V International

Join us in Munich, Germany at RISC-V Summit Europe
<https://riscv-europe.org/summit/2024/> from 24-28 June. Be a part of the
new wave of European computing innovation!

**Comment by aswaterman** [comment]
Right, they’re just there to improve the visual appearance and help readers see the boundaries between fields. There’s no semantic content.

**Comment by kersten1** [comment]
This is done.
